<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <meta name="author" content="Michael Baudis">
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Classifications and Ontologies - Progenetix Documentation</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
  <link href="../css/extra.css" rel="stylesheet" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Classifications and Ontologies";
    var mkdocs_page_input_path = "classifications-and-ontologies.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Progenetix Documentation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../use-cases/">Use Case Examples</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../services/">Services</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">Classifications and Ontologies</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#list-of-filters-recognized-by-different-query-endpoints">List of filters recognized by different query endpoints</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#public-ontologies-with-curie-based-syntax">Public Ontologies with CURIE-based syntax</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#private-filters">Private filters</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#diagnoses-phenotypes-and-histologies">Diagnoses, Phenotypes and Histologies</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#ncit-coding-of-tumor-samples">NCIt coding of tumor samples</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#current-ncit-sample-codes">Current NCIt sample codes</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#icd-coding-of-tumor-samples">ICD coding of tumor samples</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#current-icd-o-sample-codes">Current ICD-O sample codes</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#uberon-codes">UBERON codes</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#current-uberon-sample-codes">Current UBERON sample codes</a>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#genomic-variations-cnv-ontology">Genomic Variations (CNV Ontology)</a>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../publication-collection/">Publication Collection</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../changelog/">Changelog</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../beaconplus/">BeaconPlus</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="" href="http://progenetix.org">Progenetix Site</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="" href="http://info.baudisgroup.org">Baudisgroup</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Progenetix Documentation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Classifications and Ontologies</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>

          <div role="main">
            <div class="section">
              
                <h1 id="classifications-ontologies-and-concepts">Classifications, Ontologies and Concepts</h1>
<p>The Progenetix resource utilizes standardized diagnostic coding systems, with a
move towards hierarchical ontologies. As part of the coding process we have
developed and provide several code mapping resources through repositories, the
Progenetix website and APIs.</p>
<p>Additionally to diagnostic and other clinical concepts, Progenetix increasingly
uses hierarchical terms and concepts for the annotation and querying of technical
parameters such as platform technologies. Overall, the Progenetix resource uses a query syntax based around the <a href="https://beacon-project.io/v2/filters.html">Beacon v2 "filters"</a> concept with a <a href="https://www.w3.org/TR/2010/NOTE-curie-20101216/">CURIE</a> based syntax.</p>
<h2 id="list-of-filters-recognized-by-different-query-endpoints">List of filters recognized by different query endpoints</h2>
<h3 id="public-ontologies-with-curie-based-syntax">Public Ontologies with CURIE-based syntax</h3>
<table>
<thead>
<tr>
<th>CURIE prefix</th>
<th align="center">Code/Ontology</th>
<th>Examples</th>
</tr>
</thead>
<tbody>
<tr>
<td>NCIT</td>
<td align="center">NCIt Neoplasm <sup id="fnref:1"><a class="footnote-ref" href="#fn:1">1</a></sup></td>
<td>NCIT:C27676</td>
</tr>
<tr>
<td>HP</td>
<td align="center">HPO<sup id="fnref:2"><a class="footnote-ref" href="#fn:2">2</a></sup></td>
<td>HP:0012209</td>
</tr>
<tr>
<td>PMID</td>
<td align="center">NCBI Pubmed ID</td>
<td><a href="https://progenetix.org/services/ids/PMID:18810378">PMID:18810378</a></td>
</tr>
<tr>
<td>geo</td>
<td align="center">NCBI Gene Expression Omnibus<sup id="fnref:3"><a class="footnote-ref" href="#fn:3">3</a></sup></td>
<td><a href="https://progenetix.org/services/ids/geo:GPL6801">geo:GPL6801</a>, <a href="https://progenetix.org/services/ids/geo:GSE19399">geo:GSE19399</a>, <a href="https://progenetix.org/services/ids/geo:GSM491153">geo:GSM491153</a></td>
</tr>
<tr>
<td>arrayexpress</td>
<td align="center">EBI ArrayExpress <sup id="fnref:4"><a class="footnote-ref" href="#fn:4">4</a></sup></td>
<td>arrayexpress:E-MEXP-1008</td>
</tr>
<tr>
<td>cellosaurus</td>
<td align="center">Cellosaurus - a knowledge resource on cell lines <sup id="fnref:5"><a class="footnote-ref" href="#fn:5">5</a></sup></td>
<td>cellosaurus:CVCL_1650</td>
</tr>
<tr>
<td>Uberon</td>
<td align="center">Uberon Anatomical Ontology <sup id="fnref:6"><a class="footnote-ref" href="#fn:6">6</a></sup></td>
<td>UBERON:0000992</td>
</tr>
<tr>
<td>cbioportal</td>
<td align="center">cBioPortal<sup id="fnref:9"><a class="footnote-ref" href="#fn:9">9</a></sup></td>
<td><a href="https://progenetix.org/services/ids/cbioportal:msk_impact_2017">cbioportal:msk_impact_2017</a></td>
</tr>
</tbody>
</table>
<h3 id="private-filters">Private filters</h3>
<p>Since some classifications cannot directly be referenced, and in accordance with
the upcoming Beacon v2 concept of "private filters", Progenetix uses
additionally a set of structured non-CURIE identifiers.</p>
<p>For terms with a <code>pgx</code> prefix, the <a href="http://identifiers.org/pgx/">identifiers.org resolver</a> will </p>
<table>
<thead>
<tr>
<th>Filter prefix / local part</th>
<th align="center">Code/Ontology</th>
<th>Example</th>
</tr>
</thead>
<tbody>
<tr>
<td>pgx:icdom-...</td>
<td align="center">ICD-O 3<sup id="fnref2:7"><a class="footnote-ref" href="#fn:7">7</a></sup> Morphologies (Progenetix)</td>
<td>pgx:icdom-81703</td>
</tr>
<tr>
<td>pgx:icdot...</td>
<td align="center">ICD-O 3<sup id="fnref:7"><a class="footnote-ref" href="#fn:7">7</a></sup> Topographies(Progenetix)</td>
<td>pgx:icdot-C04.9</td>
</tr>
<tr>
<td>TCGA</td>
<td align="center">The Cancer Genome Atlas (Progenetix)<sup id="fnref:8"><a class="footnote-ref" href="#fn:8">8</a></sup></td>
<td>TCGA-000002fc-53a0-420e-b2aa-a40a358bba37</td>
</tr>
<tr>
<td>pgx:pgxcohort-...</td>
<td align="center">Progenetix cohorts <sup id="fnref:10"><a class="footnote-ref" href="#fn:10">10</a></sup></td>
<td>pgx:pgxcohort-arraymap</td>
</tr>
</tbody>
</table>
<h2 id="diagnoses-phenotypes-and-histologies">Diagnoses, Phenotypes and Histologies</h2>
<h3 id="ncit-coding-of-tumor-samples">NCIt coding of tumor samples</h3>
<ul>
<li>based on <a href="https://evs.nci.nih.gov/ftp1/NCI_Thesaurus/Neoplasm/About_Core.html">NCIt neoplasm core</a>
but now extended based on the whole "neoplasia" subtree of the NCI Thesaurus (<code>NCIT:C3262</code> and child terms)</li>
<li>first implementation of NCIt concepts mapping in January 2017, then for a subset of arrayMap samples</li>
<li>now providing ICD-O 3 &lt;=&gt; NCIt mappings through the <a href="https://github.com/progenetix/ICDOntologies">ICDOntoologies</a> mapping project with a <a href="https://progenetix.org/service-collection/ontologymaps/">front-end an API</a> on the website</li>
</ul>
<h4 id="current-ncit-sample-codes">Current NCIt sample codes</h4>
<ul>
<li><a href="https://progenetix.org/services/collations?collationTypes=NCIT&amp;output=text">progenetix.org/services/collations?collationTypes=NCIT&amp;output=text</a></li>
</ul>
<h3 id="icd-coding-of-tumor-samples">ICD coding of tumor samples</h3>
<p>The Progenetix resource primarily used the coding schemas of the _International Classification of Diseases in Oncology__ (3rd edition; "ICD-O 3"), to classify all biosamples for which experimental data is available. Users can get a list of ICD-O 3 codes in the Progenetix format <a href="http://info.progenetix.org/doc/services/collations.html">through Progenetix collations</a>.</p>
<p>The mappings used here for the ICD morphology codings (mapped to ICDMORPHOLOGY and ICDMORPHOLOGYCODE) are derived from the original source file last accessed on 2016-08-18 from <a href="https://www.who.int/standards/classifications/other-classifications/international-classification-of-diseases-for-oncology">the WHO</a>. The primary codes have been updated from the 2011 update document <a href="http://www.who.int/classifications/icd/updates/ICDO3Updates2011.pdf">ICDO3Updates2011.pdf</a>.</p>
<h4 id="current-icd-o-sample-codes">Current ICD-O sample codes</h4>
<ul>
<li>ICD-O Morphologies<ul>
<li><a href="https://progenetix.org/services/collations/?collationTypes=icdom&amp;output=text">progenetix.org/services/collations?collationTypes=icdom&amp;output=text</a></li>
</ul>
</li>
<li>ICD-O Topographies<ul>
<li><a href="https://progenetix.org/services/collations/?collationTypes=icdot&amp;method=ids&amp;output=text">progenetix.org/services/collations/?collationTypes=icdot&amp;method=ids&amp;output=text</a></li>
</ul>
</li>
</ul>
<h3 id="uberon-codes">UBERON codes</h3>
<p>The organ sites of the original coding have been mapped to UBERON. The mappings
are detailed in the related <a href="https://github.com/progenetix/icdot2uberon">icdot2uberon</a> project.</p>
<h4 id="current-uberon-sample-codes">Current UBERON sample codes</h4>
<ul>
<li><a href="https://progenetix.org/services/collations?collationTypes=UBERON&amp;method=counts&amp;output=text">progenetix.org/services/collations?collationTypes=UBERON&amp;method=counts&amp;output=text</a></li>
</ul>
<hr />
<h2 id="genomic-variations-cnv-ontology">Genomic Variations (CNV Ontology)</h2>
<p><img src="../img/form-structural-variant-type-selector.png" style="float: right; width: 201px; margin-top: -15px; margin-left: 10px;"/>The Progenetix repository contains predominantly copy number variants. While we
had limited CNV type annotations to the "minimum information content" - i.e. using
<code>DUP</code> and <code>DEL</code> categories for indicating relative genomic copy number gains or losses,
respectively, from 2022 Progenetix will move to a richer CNV classification in line
with "common use practices". As part of the <a href="http://cnvar.org">ELIXIR h-CNV community</a> and contributors
to the GA4GH <a href="http://genomebeacons.org">Beacon project</a> and <a href="http://vrs.org">Variant Representation Specification (VRS)</a>
we have co-developed a "CNV assessment ontology" which in January 2022 has been
accepted into the <a href="https://www.ebi.ac.uk/ols/ontologies/efo">Experimental Factor Ontology (EFO)</a>
and is under discussion at <a href="https://github.com/The-Sequence-Ontology/SO-Ontologies/issues/568">Sequence Ontology (SO)</a>
and for use in VRS.</p>
<p>In January 2022 we switched the internal representation of CNV states to EFO codes
and implemented the respective search functionality in the <code>bycon</code> package. Future
data updates will gradually add the more granular classes such as <code>EFO:0030073</code>
where they apply.</p>
<pre><code>id: EFO:0030063
label: copy number assessment
  |
  |-id: EFO:0030064
  | label: regional base ploidy
  |   |
  |   |-id: EFO:0030065
  |     label: copy-neutral loss of heterozygosity
  |
  |-id: EFO:0030066
    label: relative copy number variation
      |
      |-id: EFO:0030067
      | label: copy number loss
      |   |
      |   |-id: EFO:0030068
      |   | label: low-level copy number loss
      |   |
      |   |-id: EFO:0030069
      |     label: complete genomic deletion
      |
      |-id: EFO:0030070
        label: copy number gain
          |
          |-id: EFO:0030071
          | label: low-level copy number gain
          |
          |-id: EFO:0030072
             label: high-level copy number gain
             note: commonly but not consistently used for &gt;=5 copies on a bi-allelic genome region
              |
              |-id: EFO:0030073
                 label: focal genome amplification
                 note: &gt;-
                   commonly used for localized multi-copy genome amplification events where the
                   region does not extend &gt;3Mb (varying 1-5Mb) and may exist in a large number of
                   copies
</code></pre>
<hr />
<div class="footnote">
<hr />
<ol>
<li id="fn:1">
<p>National Cancer Institute Thesaurus Neoplasm <a href="https://bioportal.bioontology.org/ontologies/NCIT_NEOPLASM">NCIt Neoplasm</a>&#160;<a class="footnote-backref" href="#fnref:1" title="Jump back to footnote 1 in the text">&#8617;</a></p>
</li>
<li id="fn:2">
<p>Human phenotype ontology <a href="https://hpo.jax.org">HPO</a>&#160;<a class="footnote-backref" href="#fnref:2" title="Jump back to footnote 2 in the text">&#8617;</a></p>
</li>
<li id="fn:3">
<p>Supported identifiers include platforms(GPL), series(GSE) and samples(GSM).<a href="https://www.ncbi.nlm.nih.gov/geo/info/overview.html">GEO Overview</a>&#160;<a class="footnote-backref" href="#fnref:3" title="Jump back to footnote 3 in the text">&#8617;</a></p>
</li>
<li id="fn:4">
<p>Supports ArrayExpress Accession ID. <a href="https://www.ebi.ac.uk/arrayexpress/browse.html">ArrayExpress browse</a>&#160;<a class="footnote-backref" href="#fnref:4" title="Jump back to footnote 4 in the text">&#8617;</a></p>
</li>
<li id="fn:5">
<p><a href="https://web.expasy.org/cellosaurus/">Cellosaurus</a> accession ID.&#160;<a class="footnote-backref" href="#fnref:5" title="Jump back to footnote 5 in the text">&#8617;</a></p>
</li>
<li id="fn:6">
<p><a href="http://uberon.github.io/about.html">Uberon</a> ID&#160;<a class="footnote-backref" href="#fnref:6" title="Jump back to footnote 6 in the text">&#8617;</a></p>
</li>
<li id="fn:7">
<p>International Classification of Diseases for Oncology, 3rd Edition <a href="https://www.who.int/standards/classifications/other-classifications/international-classification-of-diseases-for-oncology">ICD-O-3</a>&#160;<a class="footnote-backref" href="#fnref:7" title="Jump back to footnote 7 in the text">&#8617;</a><a class="footnote-backref" href="#fnref2:7" title="Jump back to footnote 7 in the text">&#8617;</a></p>
</li>
<li id="fn:8">
<p>Supports <a href="https://portal.gdc.cancer.gov">TCGA</a> Sample UUID.&#160;<a class="footnote-backref" href="#fnref:8" title="Jump back to footnote 8 in the text">&#8617;</a></p>
</li>
<li id="fn:9">
<p>Supports <a href="https://www.cbioportal.org/datasets">cBioPortal</a> Study ID.&#160;<a class="footnote-backref" href="#fnref:9" title="Jump back to footnote 9 in the text">&#8617;</a></p>
</li>
<li id="fn:10">
<p>Cohorts defined in Progenetix involving a collection of related samples. Currently includes (add <code>pgxcohort-</code>): arraymap, 2021progenetix, DIPG, TCGA, TCGAcancers, <a href="https://progenetix.org/progenetix-cohorts/gao-2021-signatures/">gao2021signatures</a>.&#160;<a class="footnote-backref" href="#fnref:10" title="Jump back to footnote 10 in the text">&#8617;</a></p>
</li>
</ol>
</div>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../publication-collection/" class="btn btn-neutral float-right" title="Publication Collection">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../services/" class="btn btn-neutral" title="Services"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
      <p>&copy; Copyright 2022, Michael Baudis and progenetix.org</p>
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
    
      <span><a href="../services/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../publication-collection/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme_extra.js" defer></script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
