<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <meta name="author" content="Michael Baudis">
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Services - Progenetix Documentation</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
  <link href="../css/extra.css" rel="stylesheet" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Services";
    var mkdocs_page_input_path = "services.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Progenetix Documentation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../use-cases/">Use Case Examples</a>
                    </li>
                </ul>
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">Services</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#formats">Formats</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#servicespy-and-url-mapping">services.py and URL Mapping</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#api-response-formats">API Response formats</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#data-file-formats-pgxseg-segments">Data File Formats - .pgxseg Segments</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#pgxseg-sample-segment-files">.pgxseg Sample Segment Files</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#pgxseg-segment-cnv-frequencies">.pgxseg Segment CNV Frequencies</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#data-matrix-files">Data Matrix Files</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#cnv-frequency-matrix">CNV Frequency Matrix</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#cnv-status-matrix">CNV Status Matrix</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#image-formats">Image Formats</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#the-pgx-plotting-library">The PGX plotting library</a>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#services">Services</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#cancer-genomics-publications-publications">Cancer Genomics Publications publications</a>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#cytoband-mapping-cytomapper">Cytoband Mapping cytomapper</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#examples_2">Examples</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#response">Response</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#gene-coordinates-genespans">Gene Coordinates genespans</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#examples_3">Examples</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#ontology-cross-mapping-ontologymaps">Ontology Cross-Mapping (ontologymaps)</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#icd-o-representation">ICD-O Representation</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#parameters">Parameters</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#examples_4">Examples</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#more-information">More Information</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#public-and-local-identifiers-ids">Public and Local Identifiers ids</a>
    </li>
        </ul>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../classifications-and-ontologies/">Classifications and Ontologies</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../publication-collection/">Publication Collection</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../changelog/">Changelog</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../beaconplus/">BeaconPlus</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="" href="http://progenetix.org">Progenetix Site</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="" href="http://info.baudisgroup.org">Baudisgroup</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Progenetix Documentation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Services</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>

          <div role="main">
            <div class="section">
              
                <h1 id="progenetix-services">Progenetix Services</h1>
<p>The <em>bycon</em> environment provides a number of data services which make use of
resources in the <em>Progenetix</em> environment.</p>
<h2 id="formats">Formats</h2>
<h3 id="servicespy-and-url-mapping"><code>services.py</code> and URL Mapping</h3>
<p>The service URL format <code>progenetix.org/services/__service-name__/?parameter=value</code>
is a shorthand for <code>progenetix.org/cgi-bin/bycon/services/__service-name__.py?parameter=value</code>.</p>
<hr />
<h3 id="api-response-formats">API Response formats</h3>
<p>Standard responses are provided as <code>Content-Type: application/json</code>. The wrapper
format for JSON encoded data follows the standard Beacon response
format where the main data is usually contained in the <code>response.results</code> list.</p>
<hr />
<h3 id="data-file-formats-pgxseg-segments">Data File Formats - <code>.pgxseg</code> Segments</h3>
<p>Progenetix uses a variation of a standard tab-separated columnar text file such
as produced by array or sequencing CNV software, with an optional metadata header
for e.g. plot or grouping instructions.</p>
<p>Wile the first edition only was geared towards sample-linked segment annotations,
a variation is now being provided for CNV frequencies.</p>
<h4 id="pgxseg-sample-segment-files"><code>.pgxseg</code> Sample Segment Files</h4>
<ul>
<li>a standard tab-delimited Progenetix segments file<ul>
<li>an additional header may exist</li>
<li>only first 5 columns are necessary</li>
<li>column 5 (mean) can be empty or dot, if column 6 exists and contains status value</li>
<li>undefined fields in existing columns are replaced with the "." character</li>
</ul>
</li>
<li>header (optional)<ul>
<li>header lines start with the <code>#</code> character</li>
<li>Plot parameters:<ul>
<li>lines start with <code>#plotpars=&gt;</code></li>
<li>parameters are added in <code>parameter_name=value;other_parameter=itsValue</code> format - see below</li>
<li>basically any <a href="https://github.com/progenetix/PGX/blob/master/config/plotdefaults.yaml">plot parameter from PGX</a> can be used</li>
</ul>
</li>
<li>Sample / grouping parameters<ul>
<li>the <code>biosample_id</code> parameter is required to assign values (e.g. group labels) to samples</li>
<li><code>biosample_id</code> has to correspond to the identifiers used in column 1 of the following segments data</li>
<li><code>parameter=value</code> pairs are semicolon-separated</li>
<li>values may be wrapped in double quotation marks (<code>group_label="Ductal Breast Carcinoma"</code>)</li>
<li><code>group_id</code> <strong>should</strong> be used for grouping</li>
</ul>
</li>
<li>this is a convention for the Progenetix plotting engine</li>
<li><code>group_label</code> is optional for grouping / labeling of the groups</li>
<li>Metadata<ul>
<li>lines start with <code>#meta=&gt;</code></li>
<li>additional information about the file</li>
<li>(so far) only informative</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>For example, this API call retireves the variants for 78 samples from two NCIt
cancer types (please be aware of the <code>&amp;filterLogic=OR</code> pragma!):</p>
<ul>
<li><a href="https://progenetix.org/beacon/variants/?filters=NCIT:C6958,NCIT:C4504&amp;filterLogic=OR&amp;output=pgxseg">progenetix.org/beacon/variants/?filters=NCIT:C6958,NCIT:C4504&amp;filterLogic=OR&amp;output=pgxseg</a></li>
</ul>
<p>An excerpt of the segment file would look like below:</p>
<pre><code>#meta=&gt;biosample_count=78
#plotpars=&gt;title=&quot;Testing Custom Plot Parameters&quot;
#plotpars=&gt;subtitle=&quot;Some Chromosomes, Colors etc.&quot;
#plotpars=&gt;chr2plot=&quot;3,5,7,8,11,13,16&quot;
#plotpars=&gt;color_var_dup_hex=#EE4500;color_var_del_hex=#09F911
#plotpars=&gt;size_title_left_px=300
#plotpars=&gt;size_text_title_left_px=10
#sample=&gt;biosample_id=pgxbs-kftvhhmm;group_id=NCIT:C6393;group_label=&quot;Invasive Ductal and Invasive Lobular Breast Carcinoma&quot;
#sample=&gt;biosample_id=GSM252886;group_id=NCIT:C4504;group_label=&quot;Malignant Breast Phyllodes Tumor&quot;
biosample_id  chro  start stop  mean  variant_type  probes
pgxbs-kftvhhmm  1 911484  11993973  -0.4486 DEL .
pgxbs-kftvhhmm  1 12158755  22246766  0.2859 DUP  .
pgxbs-kftvhhmm  1 22346353  24149880  -0.5713 DEL .
pgxbs-kftvhhmm  1 24160170  33603123  0.0812  . .
pgxbs-kftvhhmm  1 33683474  37248987  -0.6478 DEL .
pgxbs-kftvhhmm  1 37391587  248655165 0.0342  . .
pgxbs-kftvhhmm  2 110819  240942225 -0.0007 . .
pgxbs-kftvhhmm  3 119131  4655519 -0.0122 . .
pgxbs-kftvhhmm  3 4662952 4857477 0.9273 DUP  .
...
</code></pre>
<h4 id="pgxseg-segment-cnv-frequencies"><code>.pgxseg</code> Segment CNV Frequencies</h4>
<p>In the frequency file</p>
<ul>
<li><code>group_id</code> values replace the <code>biosample_id</code><ul>
<li>multiple groups can be concatenated in the file</li>
</ul>
</li>
<li><code>chro</code>, <code>start</code> and <code>end</code> are the same as in the sample files</li>
<li><code>gain_frequency</code> and <code>loss_frequency</code> indicate the <em>percent</em> values for gains and losses overlapping the segment, respectively</li>
</ul>
<p>Future options are under evaluation.</p>
<p>Examples can be derived from the Progenetix "Services" API:</p>
<ul>
<li><a href="https://progenetix.org/services/intervalFrequencies/pgxcohort-TCGAcancers/?output=pgxseg">/services/intervalFrequencies/pgxcohort-TCGAcancers/?output=pgxseg</a><ul>
<li>single group in REST syntax (here overall CNV frequencies in &gt;11000 cancer samples from the TCGA sample collection)</li>
</ul>
</li>
<li><a href="https://progenetix.org/services/intervalFrequencies/?filters=icdom-81403,icdom-81443&amp;output=pgxseg">/services/intervalFrequencies/?filters=icdom-81403,icdom-81443&amp;output=pgxseg</a><ul>
<li>2 sets using the <code>filters</code> parameter</li>
</ul>
</li>
</ul>
<pre><code>#meta=&gt;genome_binning=1Mb;interval_number=3102
#group=&gt;group_id=icdom-81403;label=Adenocarcinoma, NOS;dataset_id=progenetix;sample_count=18559
group_id  chro  start end gain_frequency  loss_frequency  index
icdom-81403 1 0 1000000 8.8 9.12  0
icdom-81403 1 1000000 2000000 8.49  8.68  1
icdom-81403 1 2000000 3000000 9.81  13.19 2
icdom-81403 1 3000000 4000000 10.02 15.84 3
icdom-81403 1 4000000 5000000 7.94  15.91 4
...
icdom-81403 2 228000000 229000000 7.37  6.62  477
icdom-81403 2 229000000 230000000 7.39  6.89  478
icdom-81403 2 230000000 231000000 8.3 7.0 479
icdom-81403 2 231000000 232000000 8.24  6.86  480
icdom-81403 2 232000000 233000000 9.1 7.89  481
...
</code></pre>
<hr />
<h3 id="data-matrix-files">Data Matrix Files</h3>
<h4 id="cnv-frequency-matrix">CNV Frequency Matrix</h4>
<p>The CNV frequency matrix contains interval CNV frequencies for genomic bins, separate for gain and loss frquencies:</p>
<ul>
<li>header similar to segment frequency files</li>
<li>first column with group identifier</li>
<li>standard genome binning on GRCh38 results in 2 x 3102 value columns</li>
<li>header line indicates genomic ranges for the bins</li>
<li>first all gain frequencies (in %), then all losses</li>
</ul>
<pre><code>#meta=&gt;genome_binning=1Mb;interval_number=3102
#group=&gt;group_id=NCIT:C7376;label=Pleural Malignant Mesothelioma;dataset_id=progenetix;sample_count=240
#group=&gt;group_id=PMID:22824167;label=Beleut M et al. (2012)...;dataset_id=progenetix;sample_count=159
group_id  1:0-1000000:gainF 1:1000000-2000000:gainF ...  1:0-1000000:lossF  1:1000000-2000000:lossF ...
NCIT:C7376  9.58  7.92  ...  1.89 1.89  ...
PMID:22824167 6.29  0.0 ... 8.18  4.4 ...
</code></pre>
<h5 id="examples">Examples</h5>
<ul>
<li><a href="https://progenetix.org/services/intervalFrequencies/?datasetIds=progenetix&amp;output=pgxmatrix&amp;filters=NCIT:C7376,PMID:22824167">https://progenetix.org/services/intervalFrequencies/?datasetIds=progenetix&amp;output=pgxmatrix&amp;filters=NCIT:C7376,PMID:22824167</a></li>
</ul>
<h4 id="cnv-status-matrix">CNV Status Matrix</h4>
<p>For endpoints with per biosample or callset / analysis delvery, the Progenetix
API offers the delivery of a binned status matrix. This matrix can e.g. directly
be used for clustering CNV patterns.</p>
<ul>
<li>id columns, followed by<ul>
<li>all "gain status" columns (e.g. 3102, see above), followed by</li>
<li>all "loss status" columns</li>
</ul>
</li>
<li>the status is indicated by a coverage value, i.e. the fraction of how much the
binned interval overlaps with one or more CNVs of the given type.</li>
</ul>
<p>The header will contain sample specific information.</p>
<pre><code>#meta=&gt;id=progenetix
#meta=&gt;assemblyId=GRCh38
#meta=&gt;filters=NCIT:C4443
#meta=&gt;genome_binning=1Mb;interval_number=3102
#meta=&gt;no_info_columns=3;no_interval_columns=6204
#sample=&gt;biosample_id=pgxbs-kftvktaz;analysis_ids=pgxcs-kftwu9ca;group_id=NCIT:C6650;group_label=Ampulla of Vater adenocarcinoma;NCIT::id=NCIT:C6650;NCIT::label=Ampulla of Vater adenocarcinoma
#sample=&gt;biosample_id=pgxbs-kftvkyeq;analysis_ids=pgxcs-kftwvv3p;group_id=NCIT:C3908;group_label=Ampulla of Vater Carcinoma;NCIT::id=NCIT:C3908;NCIT::label=Ampulla of Vater Carcinoma
...
#meta=&gt;biosampleCount=26;analysisCount=26
analysis_id biosample_id  group_id  1:0-1000000:DUP 1:1000000-2000000:DUP 1:2000000-3000000:DUP 1:3000000-4000000:DUP  ...
pgxcs-kftwu9ca  pgxbs-kftvktaz  NCIT:C6650  0 0.3434  1.0 1.0
pgxcs-kftwwbry  pgxbs-kftvkzwp  NCIT:C3908  0.5801  0 0.6415  1.0
...
</code></pre>
<h5 id="examples_1">Examples</h5>
<ul>
<li><a href="https://progenetix.org/beacon/biosamples/?output=pgxmatrix&amp;filters=NCIT:C4443">progenetix.org/beacon/biosamples/?output=pgxmatrix&amp;filters=NCIT:C4443</a></li>
</ul>
<hr />
<h3 id="image-formats">Image Formats</h3>
<p>The standard format for (plot-)images generated on Progenetix is Scalable Vector Graphics (<a href="https://en.wikipedia.org/wiki/Scalable_Vector_Graphics">SVG</a>). As the name implies, SVG is <em>scalable</em>, i.e. images can be scaled up without loosing quality or expanding in storage size. However, some of teh generated images use also embedded rastered components which will deteriorate during scaling - this is e.g. the case for array probe plots.</p>
<p><code>{admonition} [Wikipedia](https://en.wikipedia.org/wiki/Scalable_Vector_Graphics)
  All major modern web browsers—including Mozilla Firefox, Internet Explorer, Google Chrome, Opera, Safari, and Microsoft Edge—have SVG rendering support.</code></p>
<p>On most pages where plots are being displayed there is a download option for the images - (please alert us where those are missing). Browsers also have the option to export SVGs themselves e.g. as PDF.</p>
<h4 id="the-pgx-plotting-library">The PGX plotting library</h4>
<p>Plots on Progenetix are generated using the <a href="http://github.com/progenetix/PGX/">PGX package</a>, a set of Perl libraries for processing and graphical representation of CNV data. The package contains tools to</p>
<ul>
<li>read and write e.g. <a href="/doc/fileformats.html">Progentix segment files</a></li>
<li>generate binned status maps</li>
<li>render plots of sample-specific and aggregate CNV data, such as histograms and CNV frequency heatmaps</li>
</ul>
<h2 id="services">Services</h2>
<h3 id="cancer-genomics-publications-publications">Cancer Genomics Publications <code>publications</code></h3>
<p>The <code>publications</code> service serves as backend API for the display of genome
screening publications through the Progenetix <a href="https://progenetix.org/publications/">Publications DB</a>.</p>
<p>It provides articles describing whole genome screening (WGS, WES, aCGH, cCGH) experiments in cancer, including some information about e.g. the numbers of samples analysed with a given technology and if sample profiles are available in Progenetix.</p>
<p>Please contact us to alert us about additional articles you are aware of. The inclusion criteria are described in the <a href="/publication-collection">documentation</a>.</p>
<p>Since 2021 you can now directly submit suggestions for matching publications to the <a href="https://github.com/progenetix/oncopubs">oncopubs repository on Github</a>.</p>
<hr />
<h3 id="cytoband-mapping-cytomapper">Cytoband Mapping <code>cytomapper</code></h3>
<p>This services parses either:</p>
<ul>
<li>a properly formatted cytoband annotation (<code>assemblyId</code>, <code>cytoBands</code>)<ul>
<li>"8", "9p11q21", "8q", "1p12qter"</li>
</ul>
</li>
<li>a concatenated <code>chroBases</code> parameter<ul>
<li><code>7:23028447-45000000</code></li>
<li><code>X:99202660</code></li>
</ul>
</li>
</ul>
<!--
While the return object is JSON by default, specifying `text=1`, together with the `cytoBands` or
`chroBases` parameter will return the text version of the opposite.
-->

<p>There is a fallback to <em>GRCh38</em> if no assembly is being provided.</p>
<p>The <code>cytobands</code> and <code>chrobases</code> parameters can be used for running the script on the command line
(see examples below). Please be aware of the "chrobases" (command line) versus "chroBases" (cgi) use.</p>
<h4 id="examples_2">Examples</h4>
<ul>
<li>retrieve coordinates for some bands on chromosome 8  <ul>
<li><a href="https://progenetix.org/services/cytomapper?assemblyId=NCBI36.1&amp;cytoBands=8q24.1">progenetix.org/services/cytomapper?assemblyId=NCBI36.1&amp;cytoBands=8q24.1</a></li>
</ul>
</li>
<li>as above, just as text:<ul>
<li><a href="https://progenetix.org/services/cytomapper?assemblyId=NCBI.1&amp;cytoBands=8q&amp;output=text">progenetix.org/services/cytomapper?assemblyId=NCBI.1&amp;cytoBands=8q&amp;output=text</a></li>
<li><em>cytomapper shortcut</em>: <a href="https://progenetix.org/services/cytomapper/?assemblyId=NCBI36.1&amp;cytoBands=8q&amp;output=text">progenetix.org/services/cytomapper/?assemblyId=NCBI36.1&amp;cytoBands=8q&amp;output=text</a></li>
</ul>
</li>
<li>get the cytobands whith which a base range on chromosome 17 overlaps, in short and long form<ul>
<li><a href="https://progenetix.org/services/cytomapper?assemblyId=GRCh37&amp;chroBases=17:800000-24326000">progenetix.org/services/cytomapper?assemblyId=GRCh37&amp;chroBases=17:800000-24326000</a></li>
</ul>
</li>
<li>using <code>curl</code> to get the text format mapping of a cytoband range, using the API <code>services</code> shortcut:<ul>
<li><code>curl -k https://progenetix.org/services/cytomapper?cytoBands\=8q21q24.1&amp;assemblyId\=hg18&amp;text\=1</code></li>
</ul>
</li>
<li>command line version of the above<ul>
<li><code>bin/cytomapper.py --chrobases 17:800000-24326000 -g NCBI36</code></li>
<li><code>bin/cytomapper.py -b 17:800000-24326000</code></li>
<li><code>bin/cytomapper.py --cytobands 9p11q21 -g GRCh38</code></li>
<li><code>bin/cytomapper.py -c Xpterq24</code></li>
</ul>
</li>
</ul>
<h4 id="response">Response</h4>
<p>As in other <strong>bycon</strong> <code>services</code>, API responses are in JSON format with the main
content being contained in the <code>response.results</code> field.</p>
<hr />
<h3 id="gene-coordinates-genespans">Gene Coordinates <code>genespans</code></h3>
<ul>
<li>genomic mappings of gene coordinats</li>
<li>initially limited to <em>GRCh38</em> and overall CDS extension</li>
<li>responds to (start-anchored) text input of HUGO gene symbols using the <code>geneSymbol</code>
parameter or path value</li>
<li>returns a list of matching gene objects (see below under <strong>Response Formats</strong>)</li>
<li>the <code>filterPrecision=exact</code> query parameter restricts the response to a single
exact gene symbol match</li>
</ul>
<h4 id="examples_3">Examples</h4>
<ul>
<li><a href="https://progenetix.org/services/genespans/?geneSymbol=TP53&amp;filterPrecision=exact">progenetix.org/services/genespans/?geneSymbol=TP53&amp;filterPrecision=exact</a></li>
<li><a href="https://progenetix.org/services/genespans/CDKN2A">progenetix.org/services/genespans/CDKN2A</a></li>
</ul>
<hr />
<h3 id="ontology-cross-mapping-ontologymaps">Ontology Cross-Mapping (<code>ontologymaps</code>)</h3>
<p>The <code>ontologymaps</code> service provides equivalency mapping between ICD-O and other
classification systems, notably NCIt. The mappings are represented in the <a href="https://github.com/progenetix/ICDOntologies">ICDOntologies</a> project and accessible trough a front-end in the <a href="https://progenetix.org/service-collection/ontologymaps">Progenetix Services area</a>.</p>
<h4 id="icd-o-representation">ICD-O Representation</h4>
<p>Our resources use an internal representation of ICD-O 3 codes since no official CURIES are provided by the IARC. The syntax is:</p>
<ul>
<li>ICD-O 3 morphologies<ul>
<li>"pgx:icdom-"<code>s/\///</code>; i.e. number only code</li>
<li>"8500/3" =&gt; <code>pgx:icdom-85003</code></li>
</ul>
</li>
<li>ICD-O 3 Topographies<ul>
<li>"icdot-" + code</li>
<li>"C53.9" =&gt; <code>pgx:icdot-C53.9</code></li>
</ul>
</li>
</ul>
<h4 id="parameters">Parameters</h4>
<h5 id="filters"><code>filters</code></h5>
<ul>
<li>required</li>
<li>comma-concatenated <strong>complete</strong> codes and/or prefixes</li>
<li>partial codes (see above for ICD-O syntax) will not be matched unless a relaxed filter precision is indicated</li>
</ul>
<h5 id="filterprecision"><code>filterPrecision</code></h5>
<ul>
<li>optional</li>
<li>to allow partial code matches (see examples below)</li>
</ul>
<h4 id="examples_4">Examples</h4>
<h5 id="ncit-and-icd-o-3">NCIt and ICD-O 3</h5>
<ul>
<li><a href="https://progenetix.org/services/ontologymaps/?filters=pgx:icdom-85003">progenetix.org/services/ontologymaps/?filters=pgx:icdom-85003</a></li>
<li><a href="https://progenetix.org/services/ontologymaps/?filters=NCIT">https://progenetix.org/services/ontologymaps/?filters=NCIT</a></li>
<li><a href="https://progenetix.org/services/ontologymaps/?filters=pgx:icdom-85003,pgx:icdot-C50.9">progenetix.org/services/ontologymaps/?filters=pgx:icdom-85003,pgx:icdot-C50.9</a></li>
<li><a href="https://progenetix.org/services/ontologymaps/?filters=icdom-85,icdot-C50&amp;filterPrecision=start">progenetix.org/services/ontologymaps/?filters=icdom-85,icdot-C50&amp;filterPrecision=start</a><ul>
<li>As in the example above, but by stemmming the query parameters and providing the <code>filterPrecision=start</code> pragma, the response will now be a list of matched data objects (inputs and equivalents)</li>
</ul>
</li>
</ul>
<h5 id="uberon-and-icd-o-3-topography">UBERON and ICD-O 3 Topography</h5>
<ul>
<li><a href="https://progenetix.org/services/ontologymaps/?filters=UBERON&amp;filterPrecision=start">progenetix.org/services/ontologymaps/?filters=UBERON&amp;filterPrecision=start</a><ul>
<li>all mappings</li>
</ul>
</li>
<li><a href="https://progenetix.org/services/ontologymaps/?filters=UBERON,icdot-C0&amp;filterPrecision=start">progenetix.org/services/ontologymaps/?filters=UBERON,icdot-C0&amp;filterPrecision=start</a><ul>
<li>all <code>C0...</code> ICD-O T matches</li>
<li>limited to <code>UBERON</code> mappings since the prefix was given, too (otherwise all the NCIT mappings would also be listed for these ICD-O T code matches)</li>
</ul>
</li>
</ul>
<h4 id="more-information">More Information</h4>
<ul>
<li><a href="https://progenetix.org/service-collection/ontologymaps">Web Interface for ICD &amp; NCIT</a></li>
<li><a href="https://progenetix.org/service-collection/uberonmaps">Interface for ICD &amp; UBERON</a></li>
</ul>
<hr />
<h3 id="public-and-local-identifiers-ids">Public and Local Identifiers <code>ids</code></h3>
<p>The <code>ids</code> service forwards compatible, prefixed ids (see <a href="https://github.com/progenetix/bycon/blob/master/services/config/ids.yaml"><code>config/ids.yaml</code></a>) to specific
website endpoints. There is no check if the id exists; this is left to the web
page handling itself.</p>
<ul>
<li><a href="https://progenetix.org/services/ids/pgxbs-kftva5zv">https://progenetix.org/services/ids/pgxbs-kftva5zv</a></li>
<li><a href="https://progenetix.org/services/ids/PMID:28966033">https://progenetix.org/services/ids/PMID:28966033</a></li>
<li><a href="https://progenetix.org/services/ids/NCIT:C3262">https://progenetix.org/services/ids/NCIT:C3262</a></li>
<li><a href="https://progenetix.org/services/ids/cellosaurus:CVCL_0022">https://progenetix.org/services/ids/cellosaurus:CVCL_0022</a></li>
<li><a href="https://progenetix.org/services/ids/pgx:icdom-81703">https://progenetix.org/services/ids/pgx:icdom-81703</a></li>
</ul>
<p>The <code>pgx</code> prefix has been registered with <a href="http://identifiers.org">identifiers.org</a>
and the service can also be used to access identifiers at Progenetix.</p>
<ul>
<li><a href="https://identifiers.org/pgx:pgxbs-kftva5zv">https://identifiers.org/pgx:pgxbs-kftva5zv</a></li>
</ul>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../classifications-and-ontologies/" class="btn btn-neutral float-right" title="Classifications and Ontologies">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../use-cases/" class="btn btn-neutral" title="Use Case Examples"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
      <p>&copy; Copyright 2022, Michael Baudis and progenetix.org</p>
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
    
      <span><a href="../use-cases/" style="color: #fcfcfc">&laquo; Previous</a></span>
    
    
      <span><a href="../classifications-and-ontologies/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme_extra.js" defer></script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
