<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <meta name="author" content="Michael Baudis">
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Use Case Examples - Progenetix Documentation</title>
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700" />

  <link rel="stylesheet" href="../css/theme.css" />
  <link rel="stylesheet" href="../css/theme_extra.css" />
  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/styles/github.min.css" />
  <link href="../css/extra.css" rel="stylesheet" />
  
  <script>
    // Current page data
    var mkdocs_page_name = "Use Case Examples";
    var mkdocs_page_input_path = "use-cases.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/10.5.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Progenetix Documentation</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
                <ul class="current">
                    <li class="toctree-l1 current"><a class="reference internal current" href="./">Use Case Examples</a>
    <ul class="current">
    <li class="toctree-l2"><a class="reference internal" href="#explore-gene-cnvs">Explore Gene CNVs</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#example-procedure">Example Procedure</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#download-or-plot-cnv-frequencies">Download or Plot CNV Frequencies</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#example-procedure-download-cnv-frequencies">Example Procedure - Download CNV Frequencies</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#getting-cancer-type-cnv-frequencies">Getting cancer type CNV frequencies</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#download-the-data-file">Download the data file</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l3"><a class="reference internal" href="#example-procedure-download-or-embed-cnv-frequency-plot">Example Procedure - Download or embed CNV frequency plot</a>
        <ul>
    <li class="toctree-l4"><a class="reference internal" href="#examples_1">Examples</a>
    </li>
    <li class="toctree-l4"><a class="reference internal" href="#additional-plot-parameters">Additional plot parameters</a>
    </li>
        </ul>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#download-sample-data">Download Sample Data</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#examples_2">Examples</a>
    </li>
        </ul>
    </li>
    <li class="toctree-l2"><a class="reference internal" href="#user-provided-cnv-data">User-Provided CNV Data</a>
        <ul>
    <li class="toctree-l3"><a class="reference internal" href="#example-procedure_1">Example Procedure</a>
    </li>
        </ul>
    </li>
    </ul>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../services/">Services</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../classifications-and-ontologies/">Classifications and Ontologies</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../publication-collection/">Publication Collection</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../changelog/">Changelog</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="reference internal" href="../beaconplus/">BeaconPlus</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="" href="http://progenetix.org">Progenetix Site</a>
                    </li>
                </ul>
                <ul>
                    <li class="toctree-l1"><a class="" href="http://info.baudisgroup.org">Baudisgroup</a>
                    </li>
                </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Progenetix Documentation</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
    
    <li>Use Case Examples</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  
  <hr/>
</div>

          <div role="main">
            <div class="section">
              
                <h1 id="use-cases">Use Cases</h1>
<h2 id="explore-gene-cnvs">Explore Gene CNVs</h2>
<p>One of the main use cases for the Progenetix resource is the exploration of frequency and disease specificity of genes of interest. Traditionally, the relevance of somatic CNVs hitting a gene in the context of cancer are judged by</p>
<ul>
<li>the CNV frequency (i.e. in what fraction of samples the a CNV in this gene is being observed)</li>
<li>the relative specificity, i.e. how CNVs in this gene compare to</li>
<li>the overall amount of CNVs in the samples</li>
<li>the local specificity, i.e. the "focality" of the CNVs</li>
</ul>
<p>The <a href="https://progenetix.org/biosamples/">Progenetix Search Page</a> supports the exploration of regional CNVs through</p>
<ul>
<li>support for inserting positions of genes or cytobands into standard Beacon query parameter fields</li>
<li>selection support for hierarchical disease annotations</li>
<li>providing example queries as templates</li>
</ul>
<p>The response of the queries for genomic variants provide some basic statistics, e.g. the relative frequencies of these variants for each classification code (e.g. NCIT, ICD-O) encountered in teh matched samples.</p>
<h4 id="example-procedure">Example Procedure</h4>
<p><strong>TBD</strong></p>
<hr />
<h2 id="download-or-plot-cnv-frequencies">Download or Plot CNV Frequencies</h2>
<p>The Progenetix resource provides pre-computed CNV frequencies for all its
"collations" such as</p>
<ul>
<li>cancer types by e.g. NCIt, ICD-O morphology and topography codes</li>
<li>experimental series, e.g. all samples from a given publication</li>
<li>custom cohorts, e.g. all samples used in a Progenetix meta-analysis or
external project such as TCGA</li>
</ul>
<p>This data can be accessed through the Progenetix API in data and image format.</p>
<p><img alt="Example CNV histogram with custom parameters" src="https://progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=NCIT:C7376&amp;-size_plotarea_h_px=40&amp;-value_plot_y_max=50&amp;-colorschema=bluered&amp;-chr2plot=1,3,9,17,22" /></p>
<p>Interval frequencies are per default stored in a 1Mb binned format. More
information about the API use can be found <a href="/doc/services/intervalfrequencies.html">in the IntervalFrequencies API documentation</a>.</p>
<h3 id="example-procedure-download-cnv-frequencies">Example Procedure - Download CNV Frequencies</h3>
<p>Typical cases for the use of collation-specific frequency data could be e.g.
the visualization of CNV tracks with 3rd party tools such as <a href="http://www.circos.ca/software/">Circos</a>
or integration in data analysis workflows, e.g. for comparing target genes to
local, disease-specific CNV frequencies.</p>
<h4 id="getting-cancer-type-cnv-frequencies">Getting cancer type CNV frequencies</h4>
<p>All cancer codes for a given classification system can be retrieved though:</p>
<ul>
<li>NCIT</li>
<li><a href="https://progenetix.org/services/collations?filters=NCIT&amp;method=counts&amp;output=text">https://progenetix.org/services/collations?filters=NCIT&amp;method=counts&amp;output=text</a></li>
<li>ICD-O Morphologies</li>
<li><a href="https://progenetix.org/services/collations?filters=icdom&amp;method=counts&amp;output=text">https://progenetix.org/services/collations?filters=icdom&amp;method=counts&amp;output=text</a></li>
<li>please be aware that we have to use transformed ICD-O codes; e.g.
  "ICD-O 8500/3" is represented as <code>pgx:icdom-85003</code> (<code>s/^(\d{4})\/(\d)$/pgx:icdom-$1$2/</code>)</li>
<li>ICD-O Topographies</li>
<li><a href="https://progenetix.org/services/collations?filters=icdot&amp;method=counts&amp;output=text">https://progenetix.org/services/collations?filters=icdot&amp;method=counts&amp;output=text</a></li>
</ul>
<h4 id="download-the-data-file">Download the data file</h4>
<p>For any of those codes one can create a <code>.pgxseg</code> file downloader link for the
<a href="/doc/services/intervalfrequencies.html">"IntervalFrequencies" service</a>:</p>
<h5 id="examples">Examples</h5>
<ul>
<li>https://progenetix.org/services/intervalFrequencies/?output=pgxseg&amp;filters=NCIT:C105555</li>
<li>https://progenetix.org/services/intervalFrequencies/?output=pgxseg&amp;filters=icdom-85003</li>
</ul>
<h3 id="example-procedure-download-or-embed-cnv-frequency-plot">Example Procedure - Download or embed CNV frequency plot</h3>
<p>For the generation of CNV frequency plots, the same procedure as above for
identifying existing frequency maps can be applied. CNV hsitograms in <a href="/services.html#image-formats">SVG format</a>
can be generated for download or embedding through a canonical service URL with
added (single) collation code.</p>
<h5 id="examples_1">Examples</h5>
<ul>
<li><a href="https://progenetix.org/services/collationPlots/?id=pgx:icdom-85003">progenetix.org/services/collationPlots/?id=pgx:icdom-85003</a></li>
<li><a href="https://progenetix.org/services/collationPlots/?id=PMID:22824167">progenetix.org/services/collationPlots/?id=PMID:22824167</a></li>
</ul>
<h4 id="additional-plot-parameters">Additional plot parameters</h4>
<p>Plot parameters can be added to the request using a standard <code>&amp;-__parameter__=__value__</code>
syntax. Please be aware of the <code>-</code> prefix.</p>
<ul>
<li><code>-size_plotimage_w_px</code></li>
<li>modifies the width of the plot image in px (default 800)</li>
<li><a href="https://progenetix.org/services/collationPlots/?id=PMID:22824167&amp;-size_plotimage_w_px=1084">https://progenetix.org/services/collationPlots/?id=PMID:22824167&amp;-size_plotimage_w_px=1084</a></li>
<li><code>-size_plotarea_h_px</code></li>
<li>height of the plot area (excluding labels etc.) in px (default 100)</li>
<li><a href="https://progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=NCIT:C7376&amp;-size_plotarea_h_px=300">progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=NCIT:C7376&amp;-size_plotarea_h_px=300</a></li>
<li><code>-value_plot_y_max</code></li>
<li>modifies the histogram's maximum value in percent (default 100)</li>
<li><a href="https://progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=pgxcohort-TCGAcancers&amp;-value_plot_y_max=50">progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=pgxcohort-TCGAcancers&amp;-value_plot_y_max=50</a></li>
<li><code>-colorschema</code></li>
<li>change of colors used for gains and losses</li>
<li>options<ul>
<li><code>orangeblue</code> (default)</li>
<li><code>redgreen</code></li>
<li><code>greenred</code></li>
<li><code>bluered</code></li>
</ul>
</li>
<li><code>-chr2plot</code></li>
<li>comma-concatenated list of chromosomes to plot</li>
<li>
<p>default is 1 -&gt; 22 since X &amp; Y are not always correctly normalized for CNV
  frequencies</p>
<ul>
<li><code>-chr2plot=1,2,3,44,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,X,Y</code></li>
<li><code>-chr2plot=9</code></li>
</ul>
</li>
<li>
<p><a href="https://progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=NCIT:C7376&amp;-size_plotarea_h_px=40&amp;-value_plot_y_max=50&amp;-colorschema=bluered&amp;-chr2plot=1,3,9,17,22">progenetix.org/cgi/PGX/cgi/collationPlots.cgi?id=NCIT:C7376&amp;-size_plotarea_h_px=40&amp;-value_plot_y_max=50&amp;-colorschema=bluered&amp;-chr2plot=1,3,9,17,22</a></p>
<ul>
<li>see example above (live representation of embedded API call for this example)</li>
</ul>
</li>
</ul>
<hr />
<h2 id="download-sample-data">Download Sample Data</h2>
<p>The sample annotations for any type of query - particularly but not limited to
single identifier values - can be downloaded in either <a href="../beaconplus/#beacon-api">Beacon v2 JSON</a>
or in a tab-delimited text format (<code>&amp;output=table</code>).</p>
<h5 id="examples_2">Examples</h5>
<ul>
<li>Download all TCGA cancer samples from Progenetix as tab-delimited table</li>
<li><a href="http://progenetix.org/beacon/biosamples/?filters=pgx:cohort-TCGAcancers&amp;output=table">progenetix.org/beacon/biosamples/?filters=pgx:cohort-TCGAcancers&amp;output=table</a></li>
</ul>
<hr />
<h2 id="user-provided-cnv-data">User-Provided CNV Data</h2>
<p>The Progenetix resource has a limited option to visualize CNV data provided by the users. Data has to be formatted in a standard tab-delimited columnar format, preferably using the <a href="/doc/fileformats.html"><code>.pgxseg</code> file format</a>. Additional information can be found on the upload service page:</p>
<ul>
<li><a href="https://progenetix.org/service-collection/uploader/">Upload Files for CNV Visualization</a></li>
</ul>
<p>The data is only temporary stored on the server with a randomiized access path. At this time we do not provide any long term or login protected data storage and do not store access data.</p>
<h4 id="example-procedure_1">Example Procedure</h4>
<p><strong>TBD</strong></p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../services/" class="btn btn-neutral float-right" title="Services">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
      <p>&copy; Copyright 2022, Michael Baudis and progenetix.org</p>
    
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
    
    
      <span><a href="../services/" style="color: #fcfcfc">Next &raquo;</a></span>
    
  </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme_extra.js" defer></script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>
    <script defer>
        window.onload = function () {
            SphinxRtdTheme.Navigation.enable(true);
        };
    </script>

</body>
</html>
